src/Components/MapComponent/MapComponent.jsx

```diff
@@ -147,9 +162,112 @@ class MapComponent extends Component {
       onSetCurrentStopsGeoJSON(newCurentStopsGeoJSON);
     });
 
+    const modify = new Modify({
+      source: this.routeVectorSource,
+      pixelTolerance: 2,
+      style: () => {
+        const { currentMot } = this.props;
+        return pointStyleFunction(currentMot);
+      },
+    });
+
+    modify.on('modifystart', evt => {
+      // save start point to find where to add the new HOP!
+      this.initialRouteDrag = {
+        features: evt.features.getArray(),
+        coordinate: evt.mapBrowserEvent.coordinate,
+      };
+    });
+
+    modify.on('modifyend', evt => {
+      const { features } = this.initialRouteDrag;
+      const {
+        currentMot,
+        currentStops,
+        currentStopsGeoJSON,
+        onSetCurrentStops,
+        onSetCurrentStopsGeoJSON,
+      } = this.props;
+      const updatedCurrentStops = _.clone(currentStops);
+      const updatedCurrentStopsGeoJSON = _.clone(currentStopsGeoJSON);
+      let newHopIdx = -1;
+
+      // No drag for foot/car for now on.
+      if (!GRAPHHOPPER_MOTS.includes(currentMot)) {
+        const flatCoords = features
+          .map(f => f.getGeometry())
+          .map(lineString => {
+            return [
+              ...lineString.getFirstCoordinate(),
+              ...lineString.getLastCoordinate(),
+            ];
+          });
+
+        const closestSegment = this.routeVectorSource
+          .getClosestFeatureToCoordinate(this.initialRouteDrag.coordinate)
+          .getGeometry();
+
+        const closestEdges = [
+          ...closestSegment.getFirstCoordinate(),
+          ...closestSegment.getLastCoordinate(),
+        ];
+
+        flatCoords.forEach((segment, idx) => {
+          if (
+            segment.length === closestEdges.length &&
+            segment.sort().every((value, index) => {
+              return value === closestEdges.sort()[index];
+            })
+          ) {
+            newHopIdx = idx + 1;
+          }
```

Comment:
This part  of the code is not clear , do you try to find if the segment is the closest edges ?

Reply:
and why do you sort the array ?

Reply:
I read that sorting array before comparing it like this could be a bit more efficient, but I haven't made a benchmark for it ;)
I think you are write, for such a feature, we should rather remove it to keep the code simpler.
:+1: